storage/innobase/xtrabackup/test/t/xb_duplicate_sdi.sh

```diff
@@ -0,0 +1,44 @@
+
+############################################################################
+# PXB-2101 --prepare crash if exchange partition happened
+############################################################################
+
+. inc/common.sh
+start_server
+
+vlog "case#1 check exchange partition"
+
+mysql test <<EOF
+CREATE TABLE t1(a  INT NOT NULL, PRIMARY KEY(a)) engine = InnoDB;
+CREATE TABLE pt1 (a INT NOT NULL, PRIMARY KEY(a)) engine = InnoDB
+PARTITION BY RANGE (a) PARTITIONS 3 (
+PARTITION p1 VALUES LESS THAN (1000),
+PARTITION p2 VALUES LESS THAN (2000),
+PARTITION p3 VALUES LESS THAN (3000));
+ALTER TABLE pt1 EXCHANGE PARTITION p1 WITH TABLE t1;
+EOF
+backup_and_restore test
+
+vlog "case#2 check import tablespace"
+
+mysql test <<EOF
+DROP TABLE t1;
+CREATE TABLE t1(a  INT NOT NULL, PRIMARY KEY(a)) engine = InnoDB;
+ALTER TABLE t1 DISCARD TABLESPACE;
+CREATE DATABASE test2;
+use test2;
+CREATE TABLE t1(a  INT NOT NULL, PRIMARY KEY(a)) engine = InnoDB;
+FLUSH table t1 for EXPORT;
+UNLOCK TABLES;
+EOF
+
```

Comment:
we need a zipped datadir. Old version will have duplicate SDI (after doing exchange, import etc). Please create a backup with such tables.

Reply:
with current version, your patch isn't even executed, isn't it?

Reply:
that's why I executed it against 8.0.23 in jenkins.  let me create a zipper datadir as well

Reply:
Yes, better to have zipped datadir because, in the future, we won't specifically test with 8.0.23 version. we won't remember.

Reply:
added the zipped datadir